lua/gitlab/hunks/init.lua

```diff
@@ -0,0 +1,280 @@
+local List = require("gitlab.utils.list")
+local u = require("gitlab.utils")
+local state = require("gitlab.state")
+local M = {}
+
+---@class Hunk
+---@field old_line integer
+---@field old_range integer
+---@field new_line integer
+---@field new_range integer
+
+---@class HunksAndDiff
+---@field hunks Hunk[] list of hunks
+---@field all_diff_output table The data from the git diff command
+
+---Turn hunk line into Lua table
+---@param line table
+---@return Hunk|nil
+M.parse_possible_hunk_headers = function(line)
+  if line:sub(1, 2) == "@@" then
+    -- match:
+    --  @@ -23 +23 @@ ...
+    --  @@ -23,0 +23 @@ ...
+    --  @@ -41,0 +42,4 @@ ...
+    local old_start, old_range, new_start, new_range = line:match("@@+ %-(%d+),?(%d*) %+(%d+),?(%d*) @@+")
+
+    return {
+      old_line = tonumber(old_start),
+      old_range = tonumber(old_range) or 0,
+      new_line = tonumber(new_start),
+      new_range = tonumber(new_range) or 0,
+    }
+  end
+end
+---@param linnr number
+---@param hunk Hunk
+---@param all_diff_output table
+---@return boolean
+local line_was_removed = function(linnr, hunk, all_diff_output)
+  for matching_line_index, line in ipairs(all_diff_output) do
+    local found_hunk = M.parse_possible_hunk_headers(line)
+    if found_hunk ~= nil and vim.deep_equal(found_hunk, hunk) then
+      -- We found a matching hunk, now we need to iterate over the lines from the raw diff output
+      -- at that hunk until we reach the line we are looking for. When the indexes match we check
+      -- to see if that line is deleted or not.
+      for hunk_line_index = found_hunk.old_line, hunk.old_line + hunk.old_range, 1 do
+        local line_content = all_diff_output[matching_line_index + 1]
+        if hunk_line_index == linnr then
+          if string.match(line_content, "^%-") then
+            return true
+          end
+        end
+      end
+    end
+  end
+  return false
+end
+
+---@param linnr number
+---@param hunk Hunk
+---@param all_diff_output table
+---@return boolean
+local line_was_added = function(linnr, hunk, all_diff_output)
+  for matching_line_index, line in ipairs(all_diff_output) do
+    local found_hunk = M.parse_possible_hunk_headers(line)
+    if found_hunk ~= nil and vim.deep_equal(found_hunk, hunk) then
+      -- For added lines, we only want to iterate over the part of the diff that has has new lines,
+      -- so we skip over the old range. We then keep track of the increment to the original new line index,
+      -- and iterate until we reach the end of the total range of this hunk. If we arrive at the matching
+      -- index for the line number, we check to see if the line was added.
+      local i = 0
+      local old_range = (found_hunk.old_range == 0 and found_hunk.old_line ~= 0) and 1 or found_hunk.old_range
+      for hunk_line_index = matching_line_index + old_range + 1, matching_line_index + old_range + found_hunk.new_range, 1 do
```

Comment:
Hi, Harrison, looks like the `old_range + 1` in this line is causing another index error, similar to the `- 1` in `for hunk_line_index = found_hunk.old_line, hunk.old_line + hunk.old_range - 1, 1 do` that used to be in `line_was_removed`.

When I have this 
```text
abc    -> abc
       -> new         <- comment here succeeds with `line_content = "+added"`
       -> added       <- comment here fails with `line_content = nil` because line "def" is out of hunk
def    -> def
```

If I remove `+ 1` on line 73, then it works OK.